build(gha): Run `api docs` test only on backend changes

billyvg opened the pull request:
We should only run api docs test on backend file changes

joshuarli commented on .github/workflows/api-docs-test.yml:
```diff
@@ -18,22 +18,33 @@ jobs:
 
       - uses: volta-cli/action@v1
```
Move this after the paths-filter and guard with the if.

joshuarli reviewed (changes requested):
1 requested change, but a bigger question: checking for python file changes isn't enough here, we'd also need to check for package.json and yarn.lock for example. And if we can split off those api docs dependencies that would be really great.

billyvg commented:
@joshuarli Good catch here, I've refactored so that `api-docs` dependencies are defined in its own `package.json` and changed file filters to also include the `api-docs` directory.

billyvg commented on .github/workflows/api-docs-test.yml:
```diff
@@ -43,23 +55,17 @@ jobs:
           python: 3
           snuba: true
 
-      - name: yarn cache
-        uses: actions/cache@v2
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
-        with:
-          path: ${{ steps.setup.outputs.yarn-cache-dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
-
```
I just removed this as it's not really necessary... although I may want to followup and change what we're hashing... we should only hash the main `yarn.lock` file

joshuarli replied:
Why? Isn't it an easy change? Instead of `**/yarn.lock`, you want `api-docs/yarn.lock`.

billyvg replied:
`yarn install` currently takes 5s, doesn't seem worth the overhead?

joshuarli replied:
Oh, I see.

billyvg commented on .gitignore:
```diff
@@ -22,8 +22,7 @@ sentry-package.json
 /tmp
-/node_modules/
-/docs-ui/node_modules/
+node_modules
```
There's no reason to not ignore all node_modules

billyvg commented on package.json:
```diff
@@ -206,13 +203,14 @@
     "build-production": "NODE_ENV=production yarn webpack --mode production --display errors-only",
     "build": "yarn build-production --output-path=public",
-    "validate-api-examples": "openapi-examples-validator ./api-docs/openapi.json --no-additional-properties"
+    "validate-api-examples": "yarn install-api-docs && cd api-docs && yarn openapi-examples-validator ./openapi.json --no-additional-properties"
```
I don't really want to deal with `yarn` workspaces right now, but since there's only a small number of deps, I think this is fine.
